vars/whenChanged.groovy

```diff
@@ -0,0 +1,19 @@
+def call(String regex, Closure closure) {
```

Comment:
Can we support `startsWith(directory)` style matching too? Since this is just pointing to a directory it would be great if we could supply a string like `x-pack/legacy/plugins/siem/` and have `whenChanged()` find any file that starts with that string.

Reply:
What do you think about the current changes?

You can do `whenChanged(regex: /some-regex/){}` or `whenChanged(startsWith: 'x-pack/'){}`
and
`whenChanged('some-path/whatever') {}` defaults to `startsWith` rather than `regex`